vendor/curl/curl.odin

```diff
@@ -0,0 +1,362 @@
+  URL_Malformat           = 3,  // CURLE_URL_MALFORMAT
+  Not_Built_In            = 4,  // CURLE_NOT_BUILT_IN
+  Couldnt_Resolve_Proxy   = 5,  // CURLE_COULDNT_RESOLVE_PROXY
+  Couldnt_Resolve_Host    = 6,  // CURLE_COULDNT_RESOLVE_HOST
+  Couldnt_Connect         = 7,  // CURLE_COULDNT_CONNECT
+  Remote_Access_Denied    = 9,  // CURLE_REMOTE_ACCESS_DENIED
+  HTTP_Returned_Error     = 22, // CURLE_HTTP_RETURNED_ERROR
+  Write_Error             = 23, // CURLE_WRITE_ERROR
+  Upload_Failed           = 25, // CURLE_UPLOAD_FAILED
+  Read_Error              = 26, // CURLE_READ_ERROR
+  Out_Of_Memory           = 27, // CURLE_OUT_OF_MEMORY
+  Operation_Timedout      = 28, // CURLE_OPERATION_TIMEDOUT
+  SSL_Connect_Error       = 35, // CURLE_SSL_CONNECT_ERROR
+  Bad_Download_Resume     = 36, // CURLE_BAD_DOWNLOAD_RESUME
+  Too_Many_Redirects      = 47, // CURLE_TOO_MANY_REDIRECTS
+  Got_Nothing             = 52, // CURLE_GOT_NOTHING
+  SSL_Engine_Not_Found    = 53, // CURLE_SSL_ENGINE_NOTFOUND
+  SSL_Engine_Set_Failed   = 54, // CURLE_SSL_ENGINE_SETFAILED
+  Send_Error              = 55, // CURLE_SEND_ERROR
+  Recv_Error              = 56, // CURLE_RECV_ERROR
+  SSL_Cert_Problem        = 58, // CURLE_SSL_CERTPROBLEM
+  SSL_Cipher              = 59, // CURLE_SSL_CIPHER
+  Peer_Failed_Verify      = 60, // CURLE_PEER_FAILED_VERIFICATION
+  Bad_Content_Encoding    = 61, // CURLE_BAD_CONTENT_ENCODING
+  SSL_Engine_Init_Failed  = 66, // CURLE_SSL_ENGINE_INITFAILED
+  SSL_CRL_Bad_File        = 82, // CURLE_SSL_CRL_BADFILE
+  SSL_Issuer_Error        = 83, // CURLE_SSL_ISSUER_ERROR
+  SSL_Invalid_Cert_Status = 91, // CURLE_SSL_INVALIDCERTSTATUS
+  SSL_Client_Cert         = 98, // CURLE_SSL_CLIENTCERT
+}
+
+// Slist represents a linked list of strings used by curl.
+Slist :: struct {
+  // String data in this node
+  data: cstring,
+
+  // Pointer to next node
+  next: ^Slist,
+}
+
+// Context holds an easy handle and its configuration state.
+//
+// **Note**: The context must be destroyed with destroy_context when no longer needed.
+Context :: struct {
+  // The curl easy handle
+  curl: CURL,
+
+  // List of HTTP headers
+  headers: ^Slist,
+
+  // Memory allocator
+  allocator: runtime.Allocator,
+}
+
+// Write_Callback represents a callback for handling received data.
+//
+// Inputs:
+// - data: Buffer containing received data
+// - size: Size of each data unit
+// - nmemb: Number of data units
+// - userdata: User provided data pointer
+//
+// Returns: Number of bytes handled.
+Write_Callback :: #type proc "c" (data: rawptr, size: c.size_t, nmemb: c.size_t, userdata: rawptr) -> c.size_t
+
+// Import the appropriate libcurl library based on OS
+when ODIN_OS == .Windows {
+  foreign import lib "system:libcurl.dll"
+} else when ODIN_OS == .Linux {
+  foreign import lib "system:libcurl.so"
+} else when ODIN_OS == .Darwin {
+  foreign import lib "system:libcurl.dylib"
+}
+
+@(default_calling_convention="c")
+foreign lib {
```

Comment:
`@(link_prefix="curl_")`

and then remove the `curl_` prefixes from the foreign imports